soroban-sdk/src/logging.rs

```diff
@@ -61,16 +60,11 @@ use crate::{
 /// let env = Env::default();
 ///
 /// let value = 5;
-/// log!(&env, "a log entry: {}, {}", value, Symbol::short("another"));
+/// log!(&env, "a log entry", value, Symbol::short("another"));
 ///
 /// use soroban_sdk::testutils::Logger;
-///
-/// assert_eq!(
-///     env.logger().all(),
-///     std::vec![
-///         "a log entry: I32(5), Symbol(another)".to_string(),
-///     ],
-/// );
+/// let logentry = env.logger().all().last().unwrap().clone();
+/// assert!(logentry.contains("[\"a log entry\", 5, another]"));
```

Comment:
The changes to this test seem unnecessary and result in a testing approach that panics when no life are outputted rather than a comparison failure which is easier to debug.

Reply:
`impl Display for HostEvent` produces a larger thing now -- a pretty but fairly long string -- that includes the contract ID, which due to https://github.com/stellar/rs-soroban-env/issues/810 isn't a stable string, so we can at this point only do a match on it.

We can revise this, possibly by introducing a special accessor / stringifier for just the data field of the event (eg. expose the pretty-printer for ScVal) but that'll take another round of changes to the env, where the pretty-printer is. I figured this was an acceptable stopping point for this PR -- the purpose of which is just to catch up with the env change -- but I'm happy to fine-tune the way strings look in followups.